jupyter_server_fileid/manager.py

```diff
@@ -254,6 +242,45 @@ def __init__(self, *args, **kwargs):
         self.con.execute("CREATE INDEX IF NOT EXISTS ix_Files_path ON Files (path)")
         self.con.commit()
 
+    @staticmethod
+    def _normalize_separators(path):
+        """Replaces backslashes with forward slashes, removing adjacent slashes."""
+
+        parts = path.strip("\\").split("\\")
+        return "/".join(parts)
+
+    def _normalize_path(self, path):
+        """Accepts an API path and returns a "persistable" path, i.e. one prefixed
+        by root_dir that can then be persisted in a format relative to the implementation."""
+        # use commonprefix instead of commonpath, since root_dir may not be a
+        # absolute POSIX path.
+
+        norm_root_dir = self._normalize_separators(self.root_dir)
+        if posixpath.commonprefix([norm_root_dir, path]) != norm_root_dir:
+            path = posixpath.join(norm_root_dir, path)
+
+        path = self._normalize_separators(path)
+
+        return path
+
+    def _from_normalized_path(self, path: Optional[str]) -> Optional[str]:
+        """Accepts a "persistable" path and returns an API path, i.e. one relative
+        to root_dir and uses forward slashes as the path separator. Returns
+        `None` if the given path is None or is not relative to root_dir."""
+        if path is None:
+            return None
+
+        # Convert root_dir to an api path, since that's essentially what we persist.
+        norm_root_dir = self._normalize_separators(self.root_dir)
+        if posixpath.commonprefix([norm_root_dir, path]) != norm_root_dir:
+            return None
+
+        relpath = path
+        if path.startswith(norm_root_dir):
+            relpath = path[len(norm_root_dir) + 1 :]
```

Comment:
1. The `path` will always start with `norm_root_dir`, so a second conditional check isn't necessary.
2. Why not just use `posixpath.relpath()` here?